frontend/src/features/codes/IndicesList.tsx

```diff
@@ -0,0 +1,258 @@
+        onClick={(e) => {
+            e.preventDefault();
+            editFn({month: month, value: value});
+        }}
+    >
+        <span className="month">{month}</span>
+        <span className="value">{value}</span>
+    </div>
+);
+
+const LoadedIndexResultsList = ({data, editFn, currentPage}) => {
+    return (
+        <div className="results">
+            <PageCounter
+                currentPage={currentPage}
+                totalPages={data.page.total_pages}
+            />
+            <div className="list-headers">
+                <div className="list-header month">Kuukausi</div>
+                <div className="list-header value">Arvo</div>
+            </div>
+            <ul className="results-list">
+                {data?.contents.map((item: IIndex) => (
+                    <IndexListItem
+                        key={item.month}
+                        month={item.month}
+                        value={item.value as number}
+                        editFn={editFn}
+                    />
+                ))}
+            </ul>
+        </div>
+    );
+};
+
+const IndexResultList = ({setFormData, setCreateDialogOpen, indexType}) => {
+    const [currentPage, setCurrentPage] = useState(1);
+    const {data, error, isLoading} = useGetIndicesQuery({
+        indexType: indexType.label,
+        params: {
+            page: currentPage,
+            limit: 12,
+        },
+    });
+    return (
+        <div className="results">
+            <QueryStateHandler
+                data={data}
+                error={error}
+                isLoading={isLoading}
+            >
+                {!isLoading && !error ? (
+                    <>
+                        <LoadedIndexResultsList
+                            data={data}
+                            editFn={({month, value}) => {
+                                setFormData({
+                                    indexType: indexType,
+                                    month: month,
+                                    value: value,
+                                });
+                                setCreateDialogOpen(true);
+                            }}
+                            currentPage={currentPage}
+                        />
+                        <ListPageNumbers
+                            currentPage={currentPage}
+                            setCurrentPage={setCurrentPage}
+                            pageInfo={data?.page}
+                        />
+                    </>
+                ) : (
+                    <LoadingSpinner />
+                )}
+            </QueryStateHandler>
+            <></>
+        </div>
+    );
+};
+
+const IndicesList = (): JSX.Element => {
+    const [currentIndexType, setCurrentIndexType] = useState(indexTypes[0]);
+    const [editMonth, setEditMonth] = useState<string | null>(null);
+    const [editValue, setEditValue] = useState<number | null>(null);
```

Comment:
Doesn't seem like these are actually used anywhere except to set them as `null`, so they can be removed

Reply:
They are. The first use is right on the next lines:
```
const initialSaveData: IIndex = {
        indexType: indexTypes[0].label,
        month: editMonth || "",
        value: editValue || null,
    };
```
and are the way through which the Dialog form fields are populated when editing an index, instead of creating a new one.

Reply:
What i mean is, the only value they have is `null`. It's their initial value and the only value they are set to 😅 

So initialSaveData would be simplified to:
```
const initialSaveData: IIndex = {
        indexType: indexTypes[0].label,
        month: null || "",
        value: null || null,
    };
```

Reply:
I am (finally) in agreement, after testing the page without the above mentioned values (at all) - and it worked just as expected. 😅 😅
Hence, they're removed as being completely extraneous. This also makes the `initData()` function unnecessary, as it used to set the above values to null, but now only sets the `formData` variable to `initialSaveData`. So we're now directly doing that (meaning `setFormData(initialSaveData)`) where we used to call `initData()` before.